docs/frontend_architecture/testing_layout.rst

```diff
@@ -0,0 +1,186 @@
+       expect(screen.getByText("Sample Component")).toBeInTheDocument();
+     })
+
+     test('describe some use case', async () => {
+       renderComponent({
+           propA: samplePropA
+       });
+
+       // Perform some user interaction
+       await userEvent.click(screen.getByRole("button", { name: "Click" }))
+
+       expect(screen.getByText(samplePropA)).toBeInTheDocument();
+     });
+   });
+
+Some quick notes about the code snippet:
+
+1. You would need to replace the ``YourExampleComponent`` with the actual component you want to test.
+2. It is usually a good practice to have the first test as a smoke test in all your components.
+3. You can add describe blocks to add context to your tests. Remember to keep them as short as possible, and a test block should be responsible for testing only one functionality.
+4. In case if you are using some particular text/value in your assertions because they are derived from the props that you pass in, then it is advisable to explicity pass them to the ``renderComponent`` as arguments to give the reader context of where they are coming from.
+
+For example, in the ``describe some use case`` test, since we are querying on the value of ``samplePropA``, we are explicity passing it to the ``renderComponent`` even though it is the exact same as the default prop.
+
+Mocking Examples
+----------------
+
+This section lists down some code snippets that can be used as reference to see how the different dependencies can be mocked. These are just example snippets, and you would need to refactor their style to match the needs of your components.
+
+Vue Router
+~~~~~~~~~~
+
+You can pass a ``routes`` prop to the ``render`` function with the value as a new ``VueRouter`` object to mock Vue Router and to test navigation in your components. The same is included in the template by default (with no routes configured). To add additional routes to the name, you can pass it an array of objects containg the keys ``name`` and ``path`` for the required paths needed to be mocked.
+
+.. code:: javascript
+
+   const renderComponent = props => {
+     return render(YourExampleComponent, {
+       routes: new VueRouter([
+         {
+           name: "MockPageA",
+           path: "path/to/mock/page/A"
+         },
+         {
+           name: "MockPageB",
+           path: "path/to/mock/page/B"
+         }
+       ]),
+     });
+   };
+
+Store
+~~~~~
+
+A store object for a component can be mocked by simply providing the methods and properties accessed by the component as a ``store`` object. For example, in the
+`TotalPoints <https://github.com/learningequality/kolibri/blob/develop/kolibri/core/assets/src/views/TotalPoints.vue>`__ component, we need to mock some getters (``totalPoints``, ``currentUserId``, ``isUserLoggedIn``) and the ``fetchPoints`` action.
+
+The same can be done via:
+
+.. code:: javascript
+
+
+   // Helper function to render the component with Vuex store
+   const renderComponent = props => {
```

Comment:
We can do something similar to the composables mocks here, to make it clearer what are store values, and what component props. So we could change a little bit this to be something like:

```javascript
const renderComponent = props => {
     const { store = {}, ...componentProps } = props;
     return render(TotalPoints, {
       store: {
         getters: {
           totalPoints: () => store.totalPoints ?? 0,
           currentUserId: () => store.currentUserId ?? "user-01",
           isUserLoggedIn: () => store.isUserLoggedIn ?? true,
         },
         actions: {
           fetchPoints: () => store.points ?? 0,
         },
       },
       props
     });
   };
```

Also, we can delete the `routes: new VueRouter()` prop from these examples, as it is no longer required :tada:

Reply:
Got it! Thanks!